### PR TITLE
psu: Added MCU-BOOT protocol and imxrt1176 support

### PR DESCRIPTION
**Pull request by agkaminski**

### CODE REVIEW

**Comment by agkaminski** [comment]
Tested on ivt_flashloader.bin from https://github.com/JayHeng/NXP-MCUBootUtility, was able to program image to RAM

File: psu/psu.c [hunk 1]
```diff
@@ -227,6 +236,103 @@ static int sdp_errStatus(hid_device *dev)
...
+	unsigned char reserved;
+	unsigned char paramcnt;
+	int params[];
+} __attribute__((packed)) mcuboot_cmd_t;
+
+
+static unsigned short size2LE(unsigned short size)
+{
+	if (size == htons(size)) {
+		return (unsigned short)((size & (unsigned short)0xff) << 8) | (size >> 8);
```
Comment (HBuczynski) [inline psu/psu.c:259]: One-liners don't need brackets.

File: psu/psu.c [hunk 2]
```diff
@@ -227,6 +236,103 @@ static int sdp_errStatus(hid_device *dev)
...
+	unsigned char b[sizeof(mcuboot_frame_t) + sizeof(mcuboot_cmd_t) + 2 * 4] = { 0 };
+	mcuboot_frame_t *frame = (mcuboot_frame_t *)b;
+	mcuboot_cmd_t *cmd = (mcuboot_cmd_t *)(b + sizeof(mcuboot_frame_t));
+	int rc;
+
+	frame->reportID = FRAME_CMD_OUT;
+	frame->size = size2LE(sizeof(b) - sizeof(mcuboot_frame_t));
+
+	cmd->tag = MCU_GET_PROPERTY;
+	cmd->paramcnt = 2;
```
Comment (HBuczynski) [inline psu/psu.c:287]: In the documentation there is description that:
_The parameter count field in the header is always set to 2, for status code and command tag parameters_
Maybe this magic number should be a constant define ?

Comment (agkaminski) [reply psu/psu.c:287]: I think it's ok to leave it as number, we define only 2 parameters next it anyway

Comment (HBuczynski) [reply psu/psu.c:287]: Ok

File: psu/psu.c (hunk 1 above)
Comment (agkaminski) [reply psu/psu.c:259]: Fixed

**Comment by agkaminski** [comment]
Added missing endianess swap for parameters too

**Pull request merged**